packages/contracts/contracts/minter-suite/Minters/MinterSEAV1.sol

```diff
@@ -0,0 +1,1169 @@
+        // for that token ID.
+        require(
+            _auction.tokenId == _tokenId,
+            "Token ID does not match auction"
+        );
+
+        // ensure auction is not already ended
+        require(auctionEndTime > block.timestamp, "Auction already ended");
+
+        // require bid to be sufficiently greater than current highest bid
+        // @dev no overflow enforced automatically by solidity ^0.8.0
+        require(
+            msg.value >= SEALib.getMinimumNextBid(_auction),
+            "Bid is too low"
+        );
+
+        // EFFECTS
+        // record previous highest bider for refunding
+        address payable previousBidder = _auction.currentBidder;
+
+        // update auction bid state
+        SEALib.AuctionUpdateBid({
+            _auction: _auction,
+            _timeBufferSeconds: minterTimeBufferSeconds,
+            _bidAmount: msg.value,
+            _bidder: payable(msg.sender)
+        });
+
+        // INTERACTIONS
+        // refund previous highest bidder
+        SplitFundsLib.forceSafeTransferETH({
+            _to: previousBidder,
+            _amount: previousBid,
+            _minterRefundGasLimit: minterRefundGasLimit
+        });
+
+        emit AuctionBid({
+            tokenId: _tokenId,
+            coreContract: _coreContract,
+            bidder: msg.sender,
+            bidAmount: msg.value
+        });
+    }
+
+    /**
+     * @dev Internal function to initialize an auction for the next token ID
+     * on project `_projectId` with a bid of `msg.value` from `msg.sender`.
+     * This function reverts in any of the following cases:
+     *   - project is not configured on this minter
+     *   - project is configured but has not yet reached its start time
+     *   - project has a current active auction that is not settled
+     *   - insufficient bid amount (msg.value < basePrice)
+     *   - no next token has been minted for the project (artist may need to
+     *     call `tryPopulateNextToken`)
+     *   - `_targetTokenId` does not match the next token ID for the project
+     * After initializing a new auction, this function attempts to mint a new
+     * token and assign it to the project's next token slot, in preparation for
+     * a future token auction. However, if the project has reached its maximum
+     * invocations on either the core contract or minter, the next token slot
+     * for the project will remain empty.
+     * @dev This should be executed in a nonReentrant context to provide redundant
+     * protection against reentrancy.
+     */
+    function _initializeAuctionWithBid(
+        uint256 _projectId,
+        address _coreContract,
+        uint256 _targetTokenId
+    ) internal {
+        SEALib.SEAProjectConfig storage _SEAProjectConfig = _SEAProjectConfigs[
+            _coreContract
+        ][_projectId];
+        SEALib.Auction storage _auction = _SEAProjectConfig.activeAuction;
+        // CHECKS
```

Comment:
@jakerockland - This was not part of our previous SEA minter design, but thinking we might want to add an additional constraint before initializing a new token auction.

Basically, if `msg.sender` is not the project artist, we should require that the project is not paused and is active on the core contract. This is more aligned with our typical interaction pattern.

The check adds gas to every auction-initialization transaction, but seems appropriate from a real-world security design standpoint.

Reply:
(also, there are a couple ways to implement this, and I think it is already "implemented" for every auction **except the final token,** since `_tryMintTokenToNextSlot` does mint a next token through the core contract with the permissions of `msg.sender`)

Reply:
Could you clarify why we need to do this explicitly outside of the requirement already at the core contract level (e.g. if already handled for all tokens by way of the core contract requirements on msg.sender, I'm not sure I understand why we need to special case this)?

Reply:
yeah - sorry this is a weird edge case to describe - I wrote a test to describe the exact situation (linked below). It all has to do with our use of pre-minting the next token, so we don't actually rely on the core contract stopping a new mint for the last invocation.
https://github.com/ArtBlocks/artblocks-contracts/pull/871/files#r1269957682

Reply:
Per our chat offline – I think this type of behavior is WAI and we shouldn't leak into the minter abstraction more than we need to re: paused state.

Reply:
thank you for chatting!